src/sentry/api/endpoints/organization_prevent_github_repos.py

```diff
@@ -0,0 +1,138 @@
+
+    def _fetch_seer_integrated_repos(self, organization_names: list[str]) -> dict[str, list[str]]:
+        """
+        Fetch repos from Seer for the given GitHub organizations that are integrated with Seer GH app.
+
+        Returns a dict mapping GitHub org name to list of full repo names
+        """
+        path = "/v1/automation/codegen/prevent/integrated-repos"
+        body = orjson.dumps(
+            {
+                "organization_names": organization_names,
+                "provider": "github",
+            }
+        )
+
+        response = requests.post(
+            f"{settings.SEER_AUTOFIX_URL}{path}",
+            data=body,
+            headers={
+                "content-type": "application/json;charset=utf-8",
+                **sign_with_seer_secret(body),
+            },
+        )
+        response.raise_for_status()
+
+        return response.json().get("integrated_repos", {})
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        """
+        List GitHub repositories linked to the Sentry organization for Prevent AI.
+
+        Returns GitHub repos by GitHub orgs for the specified Sentry organization.
+        """
+        logger.info(
+            "PreventGitHubRepos endpoint called",
+            extra={
+                "organization_id": organization.id,
+                "organization_slug": organization.slug,
+                "organization_name": organization.name,
+                "user_id": request.user.id if request.user.is_authenticated else None,
+            },
+        )
+
+        github_org_integrations = integration_service.get_organization_integrations(
+            organization_id=organization.id,
+            providers=[IntegrationProviderSlug.GITHUB.value],
+            status=ObjectStatus.ACTIVE,
+        )
+
+        integration_map = {}
+        for org_integration in github_org_integrations:
+            integration = integration_service.get_integration(
+                integration_id=org_integration.integration_id
+            )
+            if integration:
+                integration_map[integration.name] = integration
+
+        # Call Seer API to get integrated repos with Seer GH app
+        seer_integrated_repos = {}
+        if integration_map:
+            try:
+                seer_integrated_repos = self._fetch_seer_integrated_repos(
+                    list(integration_map.keys())
+                )
+            except Exception as e:
+                logger.warning(
+                    "Failed to fetch Seer integrated repos",
+                    extra={
+                        "organization_id": organization.id,
+                        "error": str(e),
+                    },
+                )
+
+        # Determine what repos are installed in Sentry and Seer for each GitHub org
+        org_repos = []
+        for github_org_name, integration in integration_map.items():
+            installed_repos = Repository.objects.filter(
```

Comment:
this creates an [N+1 query ](https://planetscale.com/blog/what-is-n-1-query-problem-and-how-to-solve-it)for each integration. could we think about how to do to this in one query?